recipes/azure-core-cpp/meta.yaml

```diff
@@ -0,0 +1,58 @@
+{% set name = "azure-core-cpp" %}
```

Comment:
To be consistent with other packages, we should name this li azure-core

Reply:
What are you suggesting the name should be exactly? We can't call it `azure-core` because that is used by the Python SDK: https://anaconda.org/conda-forge/azure-core

Similarly with the other SDK libraries: https://anaconda.org/conda-forge/azure-storage-common

Reply:
> To be consistent with other packages, we should name this li azure-core

@xhochy Is your suggestion to name it `lib-azure-core`? If yes, what other packages follow this convention? The [AWS SDK for C++](https://github.com/conda-forge/aws-sdk-cpp-feedstock) is named `aws-sdk-cpp`

And in general, I think it would be potentially confusing if the conda-forge recipes all used different names than upstream

Reply:
I believe there is a typo in xhochy's comment. It should read

> we should name this libazure-core

If the package is just headers, config, and libraries (no CLI tools or other binaries), then it should be named libazure-core because the package contains libazure-core.so/dylib.

Examples:
 - libclang is libraries separate from clang the tool
 - libcurl is libraries separate from curl the tool
 - libopenvino just libs from the openvino project
 - libmad is libs from the mad project

Reply:
I'm open to adding the prefix `lib`. My main concerns are consistency and transparency. I think there is an advantage to having the conda recipe names exactly match the upstream names

> libclang is libraries separate from clang the tool

But for this example, the upstream maintainers have also named it [libclang](https://clang.llvm.org/doxygen/group__CINDEX.html), right? Same with [libcurl](https://curl.se/libcurl/). This seems like a slightly different situation than what we are discussing here, where the `lib` prefix name would only exist in conda-forge

My understanding is that the prefixes `-cpp` (and `-c`) communicate that the software contains no CLI tools or other binaries. Looking across the conda-forge ecosystem, the many various AWS libraries either [end in `-cpp`](https://github.com/search?q=org%3Aconda-forge+path%3Arecipe%2Fmeta.yaml+%2Fname%3A%5Csaws-.*-cpp%2F&type=code) or start with [`aws-c-`](https://github.com/search?q=org%3Aconda-forge+path%3Arecipe%2Fmeta.yaml+%2Fname%3A%5Csaws-c-.*%2F&type=code) (including the very similar [aws-sdk-cpp](https://github.com/conda-forge/aws-sdk-cpp-feedstock)). There are also Azure C libraries that either start with `azure-c` or end in `-c` (see the [recipe for azure-iot-sdk-c](https://github.com/conda-forge/azure-iot-sdk-c-feedstock/blob/main/recipe/meta.yaml) for various iterations on the naming scheme). I see that the `lib` prefix was added to libgoogle-cloud (https://github.com/conda-forge/google-cloud-cpp-feedstock/pull/83), but the `-cpp` was also dropped.

Given the existence of the recipes for `aws-c-*`, `aws-sdk-cpp`, etc. already in conda-forge, I'd argue that a name like `azure-core-cpp` already communicates that it is only a C++ library, with the added benefit that it matches the official upstream name.

Is anyone somewhat convinced by my argument above? I don't feel that strongly about it, but I wanted to at least present the counter-argument before we deviate the name from upstream. And if we do decide to add the prefix `lib`, should we drop the suffix `-cpp`?

Reply:
I agree that the `lib` prefix is excessive. Names like `azure-core-cpp` unquestionably refer to C++ libraries. Moreover this is the official name of the libraries, and we use the official names in the other cloud SDKs.

Reply:
It's not immediately apparent that the official name of the component is 'azure-core-cpp' when the components are bundled together under a project called 'azure-sdk-for-cpp'. 

[The upstream project's docs](https://github.com/Azure/azure-sdk-for-cpp/tree/main/sdk/core/azure-core) indicate that both the CMake package and vcpkg names are `azure-core-cpp`, so let's go with that because it is explicit evidence in the upstream projects docs/source that the upstream maintainers have named these packages in that way. In the future, let's lead with links instead of researching whether or not it's conda-forge convention to name C/CXX libraries with a `lib-` prefix.